layouts/default.vue

```diff
@@ -209,8 +310,70 @@
 		notificationsPanel.value.toggle(event);
 	};
 
+	// ADMIN
+	watch([ adminMode ], () => {
+		auth.storeAdminConfig();
+	});
+
+	const adminPanel = ref();
+	const toggleAdminPanel = async (event: Event) => {
+		adminPanel.value.toggle(event);
+	};
+
+	const checkImpersonation = async () => {
+		impersonationError.value = '';
+		impersonationList.value = [];
+
+		try {
+			if (!auth.isAdmin) {
+				throw new Error('You are not authorized to impersonate users');
+			}
+
+			if (!impersonateUsername.value) {
+				throw new Error('Please enter a username to impersonate');
+			}
+
+			impersonationLoading.value = true;
+			const users = await $directus.request<User[]>(customEndpoint({
+				method: 'GET',
+				path: '/users',
+				params: {
+					filter: {
+						github_username: { _eq: impersonateUsername.value },
+					},
+				},
+			}));
+
+			if (!users.length) {
+				throw new Error('User not found');
+			}
+
+			impersonationList.value = users as (User & { github_username: string })[];
```

Comment:
99.9% of the time, there's going to be only 1 match, let's auto-apply it in these cases (change the "search" text to "apply" as well).